src/domains/gps-common/03_storage_account.tf

```diff
@@ -3,37 +3,44 @@
 ##################
 
 # storage
-module "payments_receipt_sa" {
+module "gpd_archive_sa" {
   source = "git::https://github.com/pagopa/terraform-azurerm-v3.git//storage_account?ref=v7.18.0"
 
-  name                            = replace(format("%s-payments-sa", local.project), "-", "")
+  name                            = replace(format("%s-archive-sa", local.project), "-", "")
```

Comment:
so the name looks like `<prefix><env><location>gpsarchivesa`?
maybe it's better `<prefix><env><location>gpdarchivesa`